cmd/es-rollover/app/initialize/cmd.go

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) 2021 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package initialize
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/cmd/es-rollover/app"
+	"github.com/jaegertracing/jaeger/pkg/config"
+	"github.com/jaegertracing/jaeger/pkg/config/tlscfg"
+	"github.com/jaegertracing/jaeger/pkg/es"
+	"github.com/jaegertracing/jaeger/pkg/es/client"
+	"github.com/jaegertracing/jaeger/plugin/storage/es/mappings"
+)
+
+const ilmVersionSupport = 7
+
+func Command(v *viper.Viper, logger *zap.Logger) *cobra.Command {
+	cfg := &Config{}
+	tlsFlags := tlscfg.ClientFlagsConfig{Prefix: "es"}
+	command := &cobra.Command{
+		Use:   "init http://HOSTNAME:PORT",
+		Short: "creates indices and aliases",
+		Long:  "creates indices and aliases",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				return fmt.Errorf("wrong number of arguments")
+			}
+
+			cfg.InitFromViper(v)
+			tlsOpts := tlsFlags.InitFromViper(v)
+			tlsCfg, err := tlsOpts.Config(logger)
+			if err != nil {
+				return err
+			}
+			defer tlsOpts.Close()
+
+			httpClient := &http.Client{
+				Timeout: time.Duration(cfg.Timeout) * time.Second,
+				Transport: &http.Transport{
+					Proxy:           http.ProxyFromEnvironment,
+					TLSClientConfig: tlsCfg,
+				},
+			}
+			esClient := client.Client{
+				Endpoint:  args[0],
+				Client:    httpClient,
+				BasicAuth: app.BasicAuth(cfg.Username, cfg.Password),
+			}
+
+			indicesClient := client.IndicesClient{
+				Client:               esClient,
+				MasterTimeoutSeconds: cfg.Timeout,
+			}
+			clusterClient := client.ClusterClient{
+				Client: esClient,
+			}
+
+			initCommand := InitCommand{
+				IndicesClient: indicesClient,
+				ClusterClient: clusterClient,
+				Config:        *cfg,
+			}
+			return initCommand.Do()
```

Comment:
It makes sense to move the functionality from `Do` to a separate file and as well add tests. 


I think I would prefer to define all commands in the rollover main and reuse as much code as possible and then have a separate package per action with the `Do` code.